python/telegram_bot/bot.py

```
@@ -0,0 +1,192 @@
+def fetch_and_calculate_dma(scheme_name_query=None):
+    """
+    Fetches NAV data and calculates 50-DMA and 150-DMA.
+    If scheme_name_query is provided, filters for that specific fund.
+    Otherwise, filters for Direct Growth Equity schemes.
+    """
+    conn = get_db_connection()
+    
+    # Base query for schemes
+    if scheme_name_query:
+        # Search by exact or partial name
+        query_schemes = f"SELECT scheme_code, scheme_name FROM schemes WHERE scheme_name LIKE '%{scheme_name_query}%' COLLATE NOCASE"
```

Ruff (0.16.2)

[error] Possible SQL injection vector through string-based query construction

(S608)